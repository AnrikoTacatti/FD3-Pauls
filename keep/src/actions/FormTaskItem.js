

import api from '../api/api.js';
import { OPEN_FORM_TASK_ITEM_NEW, OPEN_FORM_TASK_ITEM_EDIT } from '../stores/const.js';

export const actioncloseFormNew = (data, dispatch) => {
    dispatch({ type: OPEN_FORM_TASK_ITEM_NEW, data: data });
}

export const actioncloseFormEdit = (data, dispatch) => {
    dispatch({ type: OPEN_FORM_TASK_ITEM_EDIT, data: data });
}

export const actionsetNewTaskItem = (data, dispatch) => {
    api.setNewTaskItem(data, dispatch);
}

export const setTaskItem = (data, dispatch) => {
    api.setTaskItem(data, dispatch);
}


