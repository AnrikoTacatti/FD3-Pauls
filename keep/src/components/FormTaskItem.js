import React from 'react';
import { connect } from 'react-redux';
import {
    actioncloseFormNew,
    actioncloseFormEdit,
    actionsetNewTaskItem,
    setTaskItem
} from '../actions/FormTaskItem.js';

export class FormTaskItem extends React.PureComponent {
    constructor(props) {
        super(props);
        this.state = {
            openFormTaskItemEdit: false,
            openFormTaskItemNew: false,
            inputTaskItemNameRequired: true,
            inputTaskItemTextRequired: true,
        };
        this.inputTaskItemName = React.createRef();
        this.inputTaskItemText = React.createRef();

    }

    UNSAFE_componentWillReceiveProps(newProps) {
        this.setState({ openFormTaskItemEdit: newProps.openFormTaskItemEdit });
        this.setState({ openFormTaskItemNew: newProps.openFormTaskItemNew });

    }

    closeFormNew = () => {
        let data = { active: false };
        actioncloseFormNew(data, this.props.dispatch);
    }


    closeFormEdit = () => {
        let data = { active: false };
        actioncloseFormEdit(data, this.props.dispatch);
    }

    addNewTaskItem = () => {
        if (this.inputTaskItemName.current.value == "") {
            this.setState({ inputTaskItemNameRequired: false });
        }
        else {
            this.setState({ inputTaskItemNameRequired: true });
        }
        if (this.inputTaskItemText.current.value == "") {
            this.setState({ inputTaskItemTextRequired: false });
        }
        else {
            this.setState({ inputTaskItemTextRequired: true });
        }
        if (this.inputTaskItemName.current.value != "" && this.inputTaskItemText.current.value != "") {
            let data = { keychapter: this.state.openFormTaskItemNew.keychapter, title: this.inputTaskItemName.current.value, text: this.inputTaskItemText.current.value };
            actionsetNewTaskItem(data, this.props.dispatch);
        }
    }

    editTaskItem = () => {
        let data = { title: this.inputTaskItemName.value, text: this.inputTaskItemText.value, keychapter: this.state.openFormTaskItemEdit.keychapter, keyitem: this.state.openFormTaskItemEdit.keyitem };
        setTaskItem(data, this.props.dispatch);
    }

    render() {
        //  console.log("FormTaskItem", this.props);
        // console.log("FormTaskItem", this.state);
        return (
            <React.Fragment >
                {
                    this.state.openFormTaskItemNew.activ === true && <React.Fragment>
                        <div className="form-add-task-caption form">
                            <div className="form-inner">
                                <p className="form-title">Add task item</p>
                                <input type="text" className={`task-item-name ${this.state.inputTaskItemNameRequired == false ? "error" : ""}`} name="title" placeholder="Enter task item name" required="" ref={this.inputTaskItemName} />
                                <textarea className={`task-item-text ${this.state.inputTaskItemTextRequired == false ? "error" : ""}`} name="text" cols="60" rows="5" placeholder="Enter task item text" required="" ref={this.inputTaskItemText} />
                            </div>
                            <div className="form-add-task-caption__footer">
                                <button type="button" title="Cancel" className="form-cancel btn" onClick={this.closeFormNew}>Cancel</button>
                                <button type="button" className="form-submit btn" onClick={this.addNewTaskItem} >Submit</button>
                            </div>
                        </div>
                        <div className="form-bg"></div>
                    </React.Fragment>
                }
                {
                    this.state.openFormTaskItemEdit.activ === true && <React.Fragment>
                        <div className="form-add-task-caption form">
                            <div className="form-inner">
                                <p className="form-title">Edit task item</p>
                                <input type="text" className="task-item-name" defaultValue={this.state.openFormTaskItemEdit.name} name="name" placeholder="Enter task item name" required="" ref={c => this.inputTaskItemName = c} />
                                <textarea className="task-item-text" defaultValue={this.state.openFormTaskItemEdit.text} name="text" cols="60" rows="5" placeholder="Enter task item text" required="" ref={c => this.inputTaskItemText = c} />
                            </div>
                            <div className="form-add-task-caption__footer">
                                <button type="button" title="Cancel" className="form-cancel btn" onClick={this.closeFormEdit}>Cancel</button>
                                <button type="button" className="form-submit btn" onClick={this.editTaskItem} >Submit</button>
                            </div>
                        </div>
                        <div className="form-bg"></div>
                    </React.Fragment>
                }
            </React.Fragment >
        );

    }

}

const mapStateToProps = function (state) {
    return {
        // весь раздел Redux state под именем counters будет доступен
        // данному компоненту как this.props.counters
        openFormTaskItemEdit: state.stateTaskLists.openFormTaskItemEdit,
        openFormTaskItemNew: state.stateTaskLists.openFormTaskItemNew
    };
};

export default connect(mapStateToProps)(FormTaskItem);






