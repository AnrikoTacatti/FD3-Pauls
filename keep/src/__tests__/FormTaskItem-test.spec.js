
"use strict";
import React from 'react';
import { FormTaskItem } from '../components/FormTaskItem.js';
jest.mock('../actions/FormTaskItem.js');
import {
    actioncloseFormNew,
    actioncloseFormEdit,
    actionsetNewTaskItem,
    setTaskItem
} from '../actions/FormTaskItem.js';


describe('FormTaskItem New', () => {
    let state = {
        openFormTaskItemNew: { "activ": true, "keychapter": "-Mnu71_BYp_test" },
        openFormTaskItemEdit: false,
    }

    let dispatch = jest.fn();
    actionsetNewTaskItem.mockImplementation((dispatch, data) => { return data; });
    actioncloseFormNew.mockImplementation((dispatch, data) => { return data; });
    let UNSAFE_componentWillReceiveProps = jest.spyOn(FormTaskItem.prototype, "UNSAFE_componentWillReceiveProps");

    const component = shallow(
        <FormTaskItem {...state} dispatch={dispatch} />
    );
    component.setProps({
        openFormTaskItemNew: { "activ": true, "keychapter": "-Mnu71_BYp_test" }
    });
    test('FormTaskItem New Snapshot', () => {
        expect(component).toMatchSnapshot();
    });

    test('FormTaskItem New UNSAFE_componentWillReceiveProps', () => {
        component.setProps({
            "openFormTaskItemNew": { "activ": true, "keychapter": "-Mnu71_BYp_test" }
        });
        expect(UNSAFE_componentWillReceiveProps).toHaveBeenCalled();
    });

    test('FormTaskItem New  form-title', () => {
        const title = component.find(".form-title").text();
        expect(title).toBe("Add task item");
    });
    it('ref inputTaskItemName', () => {
        expect(component.find(".task-item-name").at(0)).toHaveLength(1);
        expect(component.instance().inputTaskItemName).toBeTruthy();
    });
    it('ref inputTaskItemName', () => {
        expect(component.find(".task-item-name").at(0)).toHaveLength(1);
        expect(component.instance().inputTaskItemText).toBeTruthy();
    });


    test('FormTaskItem New cancels', () => {
        const btn = component.find(".form-cancel");
        btn.simulate('click');
        expect(actioncloseFormNew).toHaveBeenCalledWith({ active: false }, dispatch);
    });



    /*   test('FormTaskItem New filds', () => {
           component.find(".task-item-name").at(0).simulate('change', { target: { value: 'test@example.com'} });
           component.find(".task-item-text").at(0).simulate('change', 'test name');
           console.log("ref", component.instance().inputTaskItemName);
           console.log(component.find(".task-item-name").get(0));
           const btn = component.find(".form-submit");
           btn.simulate('click');
           expect(actionsetNewTaskItem).toHaveBeenCalledWith({
               keychapter: "-Mnu71_BYp_test",
               title: 'test name',
               text: 'test text'
           }, dispatch);
       });*/
    /*
        test('FormTaskItem New filds', () => {
            console.log("tect", component.instance().inputTaskItemName);
            component.instance().inputTaskItemName.current.value = 'test name';
            component.instance().inputTaskItemText.current.value = 'test name';
            console.log("tect", component.instance().inputTaskItemName);
            component.instance().addNewTaskItem();
            expect(actionsetNewTaskItem).toHaveBeenCalledWith({
                keychapter: "-Mnu71_BYp_test",
                title: 'test name',
                text: 'test text'
            }, dispatch);
        });*/

});


describe('FormTaskItem Edit', () => {
    let state = {
        openFormTaskItemNew: false,
        openFormTaskItemEdit: {
            name: 'Мои заметки № 1',
            text: 'Мои заметки № 1',
            activ: true,
            keychapter: '0',
            keyitem: '-Mnu6P7dMJTeVv8VESLh3'
        },
    }

    let dispatch = jest.fn();
    setTaskItem((dispatch, data) => { return data; });
    actioncloseFormNew.mockImplementation((dispatch, data) => { return data; });
    let UNSAFE_componentWillReceiveProps = jest.spyOn(FormTaskItem.prototype, "UNSAFE_componentWillReceiveProps");
    const component = shallow(
        <FormTaskItem {...state} dispatch={dispatch} />
    );

    component.setProps({
        openFormTaskItemNew: false,
        openFormTaskItemEdit: {
            name: 'Мои заметки № 1',
            text: 'Мои заметки № 1',
            activ: true,
            keychapter: '0',
            keyitem: '-Mnu6P7dMJTeVv8VESLh3'
        }
    });

    test('FormTaskItem Edit Snapshot', () => {
        expect(component).toMatchSnapshot();
    });

    test('FormTaskItem Edit UNSAFE_componentWillReceiveProps', () => {
        component.setProps({
            openFormTaskItemNew: false,
            openFormTaskItemEdit: {
                name: 'Мои заметки № 1',
                text: 'Мои заметки № 1',
                activ: true,
                keychapter: '0',
                keyitem: '-Mnu6P7dMJTeVv8VESLh3'
            }
        });
        expect(UNSAFE_componentWillReceiveProps).toHaveBeenCalled();
    });

    test('FormTaskItem Edit  form-title', () => {
        const title = component.find(".form-title").text();
        expect(title).toBe("Edit task item");
    });
    test('FormTaskItem Edit cancels', () => {
        const btn = component.find(".form-cancel");
        btn.simulate('click');
        expect(actioncloseFormEdit).toHaveBeenCalledWith({ active: false }, dispatch);
    });

});