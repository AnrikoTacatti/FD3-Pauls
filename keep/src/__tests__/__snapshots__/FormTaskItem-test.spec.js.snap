// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTaskItem Edit FormTaskItem Edit Snapshot 1`] = `
<Fragment>
  <div
    className="form-add-task-caption form"
  >
    <div
      className="form-inner"
    >
      <p
        className="form-title"
      >
        Edit task item
      </p>
      <input
        className="task-item-name"
        defaultValue="Мои заметки № 1"
        name="name"
        placeholder="Enter task item name"
        required=""
        type="text"
      />
      <textarea
        className="task-item-text"
        cols="60"
        defaultValue="Мои заметки № 1"
        name="text"
        placeholder="Enter task item text"
        required=""
        rows="5"
      />
    </div>
    <div
      className="form-add-task-caption__footer"
    >
      <button
        className="form-cancel btn"
        onClick={[Function]}
        title="Cancel"
        type="button"
      >
        Cancel
      </button>
      <button
        className="form-submit btn"
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
    </div>
  </div>
  <div
    className="form-bg"
  />
</Fragment>
`;

exports[`FormTaskItem New FormTaskItem New Snapshot 1`] = `
<Fragment>
  <div
    className="form-add-task-caption form"
  >
    <div
      className="form-inner"
    >
      <p
        className="form-title"
      >
        Add task item
      </p>
      <input
        className="task-item-name "
        name="title"
        placeholder="Enter task item name"
        required=""
        type="text"
      />
      <textarea
        className="task-item-text "
        cols="60"
        name="text"
        placeholder="Enter task item text"
        required=""
        rows="5"
      />
    </div>
    <div
      className="form-add-task-caption__footer"
    >
      <button
        className="form-cancel btn"
        onClick={[Function]}
        title="Cancel"
        type="button"
      >
        Cancel
      </button>
      <button
        className="form-submit btn"
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
    </div>
  </div>
  <div
    className="form-bg"
  />
</Fragment>
`;
